### ISSUE TITLE
iOS Version 13.3 breaks ssh

### ISSUE THREAD

**Issue by graehu** [blinksh/blink#890]
After updating to iOS 13.3 I can no longer connect to my home server. After inputting the server details, there’s a momentary pause followed by blink taking a new line.

It can ping the server correctly, but past that not much else is happening.

It had been flawless up to this point, I’ve tried reinstalling it and I can connect to my server via other terminal apps (terminus). Pretty weird.

**Comment by stahldom** [comment]
Noticed the same problem today - but I haven't upgraded to iOS 13.3 yet. I was hoping that the update would fix it.

**Comment by yury** [comment]
Can't reproduce that. Can you send `ssh -vvvvvv` logs here?

**Comment by graehu** [comment]
Ok, So it seems like there was an issue with the key. I just made a new key and put that on the server. I had been using that key the day before the update so I don’t know what happened to it, but it wasn’t that the server was refusing it.

I tried ssh -vvvvvvv myhost > log.txt but it was absolutely tiny compared to the output on blink. Is there a better way to capture that log?

I’m able to work again so I’m not in much of a hurry anymore.

**Comment by stahldom** [comment]
Looks like an issue with a key - which is strange as nothing has been modified since it worked. I’ll see about reconfiguring the keys soon and let you know if that fixes anything.


crypt_set_algorithms2: Set output algorithm to aes256-gcm@openssh.com
crypt_set_algorithms2: Set HMAC output algorithm to aead-gcm
crypt_set_algorithms2: Set input algorithm to aes256-gcm@openssh.com
crypt_set_algorithms2: Set HMAC input algorithm to aead-gcm
ssh_init_rekey_state: Set rekey after 4294967296 blocks
ssh_init_rekey_state: Set rekey after 4294967296 blocks
ssh_packet_client_curve25519_reply: SSH_MSG_NEWKEYS sent
ssh_packet_socket_callback: Processing 196 bytes left in socket buffer
ssh_packet_socket_callback: packet: read type 21 [len=12,padding=10,comp=1,payload=1]
ssh_packet_process: Dispatching handler for packet type 21
ssh_packet_newkeys: Received SSH_MSG_NEWKEYS
ssh_packet_newkeys: Signature verified and valid
ssh_packet_socket_callback: Processing 180 bytes left in socket buffer
ssh_packet_socket_callback: packet: read type 7 [len=160,padding=18,comp=141,payload=141]
ssh_packet_process: Dispatching handler for packet type 7
ssh_packet_ext_info: Received SSH_MSG_EXT_INFO
ssh_packet_ext_info: Follows 1 extensions
ssh_packet_ext_info: Extension: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>
ssh_connect: current state : 7
blink: setting socket keepalive: 1
packet_send2: packet: wrote [type=5, len=32, padding_size=14, comp=17, payload=17]
ssh_service_request: Sent SSH_MSG_SERVICE_REQUEST (service ssh-userauth)
ssh_packet_socket_callback: packet: read type 6 [len=32,padding=14,comp=17,payload=17]
ssh_packet_process: Dispatching handler for packet type 6
ssh_packet_service_accept: Received SSH_MSG_SERVICE_ACCEPT
packet_send2: packet: wrote [type=50, len=48, padding_size=12, comp=35, payload=35]
ssh_packet_socket_callback: packet: read type 51 [len=32,padding=16,comp=15,payload=15]
ssh_packet_process: Dispatching handler for packet type 51
ssh_packet_userauth_failure: Access denied for 'none'. Authentication that can continue: publickey
ssh_packet_userauth_failure: Access denied for 'none'. Authentication that can continue: publickey
blink: using auth methods attempt: 1 of 5
blink: import key id_rsa
ssh_pki_import_privkey_base64: Trying to decode privkey passphrase=false
ssh_key_algorithm_allowed: Checking rsa-sha2-512 with list <ssh-ed25519-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp256-cert-v01@openssh.com,rsa-sha2-512-cert-v01@openssh.com,rsa-sha2-256-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-dss-cert-v01@openssh.com,ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,ecdsa-sha2-nistp256,rsa-sha2-512,rsa-sha2-256,ssh-rsa,ssh-dss>
ssh_key_algorithm_allowed: Checking rsa-sha2-512 with list <ssh-ed25519-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp256-cert-v01@openssh.com,rsa-sha2-512-cert-v01@openssh.com,rsa-sha2-256-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-dss-cert-v01@openssh.com,ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,ecdsa-sha2-nistp256,rsa-sha2-512,rsa-sha2-256,ssh-rsa,ssh-dss>
packet_send2: packet: wrote [type=50, len=608, padding_size=11, comp=596, payload=596]
ssh_packet_socket_callback: packet: read type 51 [len=32,padding=16,comp=15,payload=15]
ssh_packet_process: Dispatching handler for packet type 51
ssh_packet_userauth_failure: Access denied for 'publickey'. Authentication that can continue: publickey
ssh_packet_userauth_failure: Access denied for 'publickey'. Authentication that can continue: publickey

**Comment by treyharris** [comment]
@graehu and @stahldom — What OS is your destination server running? Given that you both saw this not necessarily connected to Blink updates, I wonder if your server OS auto-updated SSH. Oftentimes in many OS’s, security updates that don’t require a reboot get auto-installed even if you don’t generally have auto-updating turned on, and the SSH client and server are usually in the “security” bundles that get this behavior.

**Comment by stahldom** [comment]
This one is running Ubuntu 18.04 LTS. I’m having the same problem with other servers running the same OS.

**Comment by treyharris** [comment]
> This one is running Ubuntu 18.04 LTS. I’m having the same problem with other servers running the same OS.

Interesting, I’m on Ubuntu 18.04 LTS as well and haven’t seen this, but my SSH libs haven’t updated yet. Here’s my list, you might compare:

```console
$ apt list --installed | grep -i ssh
libssh-4/bionic-updates,bionic-security,now 0.8.0~20170825.94fa1e38-1ubuntu0.5 amd64 [installed]
libssh-gcrypt-4/bionic-updates,bionic-security,now 0.8.0~20170825.94fa1e38-1ubuntu0.5 amd64 [installed,automatic]
libssh2-1/bionic,now 1.8.0-1 amd64 [installed,automatic]                           
openssh-client/bionic-updates,bionic-security,now 1:7.6p1-4ubuntu0.3 amd64 [installed]
openssh-server/bionic-updates,bionic-security,now 1:7.6p1-4ubuntu0.3 amd64 [installed]
openssh-sftp-server/bionic-updates,bionic-security,now 1:7.6p1-4ubuntu0.3 amd64 [installed]
ssh-import-id/bionic-updates,bionic-updates,now 5.7-0ubuntu1.1 all [installed]
```

**Comment by stahldom** [comment]
Using Termius I’m able to get right in —-

openssh-client/bionic-updates,bionic-security,now 1:7.6p1-4ubuntu0.3 amd64 [installed,automatic]
openssh-server/bionic-updates,bionic-security,now 1:7.6p1-4ubuntu0.3 amd64 [installed]
openssh-sftp-server/bionic-updates,bionic-security,now 1:7.6p1-4ubuntu0.3 amd64 [installed]
ssh-import-id/bionic-updates,now 5.7-0ubuntu1.1 all [installed]

**Comment by graehu** [comment]
The output that stahldom has is pretty much identical to my output of vvvvvv. It reads a little like blink can’t find/access it’s own key. May be an issue with permissions, but I’m nearly certain it’s iOS related. I was using my key right before I updated my iPad, and immediately after I couldn’t use the key anymore. I had not updated my Linux distro. (pop_os)

The funny thing is I copied the key into Terminus and it works there. :P 
Very odd.
